### ISSUE TITLE
Postconditions: `MaybeNull` and `NotNull` – `T?` is valid parameter and return type not otherwise

### ISSUE THREAD

**Issue by Uzivatel919** [dotnet/docs#25208]
From [Postconditions: `MaybeNull` and `NotNull`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis#postconditions-maybenull-and-notnull)

>You can't specify that the return value is T?. The method returns null when the sought item isn't found. Since you can't declare a T? return type, you add the MaybeNull annotation to the method return:

(Once again) `T?` can be used in generic method declaration but it will interpreted/compiled as `reference type` `nullable annotation`.

```C#
T? Method<T> ( T? t )
{
  object? x   = Method<object> (null);
  object? x1  = Method<object?> (null);
  int? x2     = Method<int?> (null);
  //int? x3     = Method<int> (null); Error  CS1503  Argument 1: cannot convert from '<null>' to 'int'
  int x4     = Method<int> (3);
  
  return default;
}
```
Example show that for `reference type` `T?` declaration is understood. Compiler obviously choses one of available options [for some reason, maybe statistics(?)].

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9ae1afa1-7c5f-862b-08f9-23cf5928953e
* Version Independent ID: 9ae2922b-5108-ed50-768b-30f08cc3fc30
* Content: [C# Reserved attributes: Nullable static analysis](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis#postconditions-maybenull-and-notnull)
* Content Source: [docs/csharp/language-reference/attributes/nullable-analysis.md](https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/attributes/nullable-analysis.md)
* Product: **dotnet-csharp**
* Technology: **csharp-language-reference**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by Uzivatel919** [comment]
Behavior for `T?` substitution is described at [Unconstrained type parameter annotations|? annotation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/unconstrained-type-parameter-annotations#-annotation). Still even if `T?` for `value type` is substituted with `T`, it is possible to use `T?` as `parameter` or `return type`.

**Comment by BillWagner** [comment]
Fixed in #25984
